scripts/build.ts

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ts-node-transpile-only
+
+import esbuild, { type BuildOptions } from "esbuild";
+import { copyFileSync, readFileSync, writeFileSync, rmSync } from "fs";
+import { globSync as glob } from "glob";
+
+const sharedOptions: BuildOptions = {
+  sourcemap: true,
+  minify: false,
+  allowOverwrite: true,
+  packages: "external",
+};
+
+async function main() {
+  // Start with a clean slate
+  rmSync("pkg", { recursive: true });
+  // Build the source code for a neutral platform as ESM
+  await esbuild.build({
+    entryPoints: glob(["./src/*.ts", "./src/**/*.ts"]),
+    outdir: "pkg/dist-src",
+    bundle: false,
+    platform: "neutral",
+    format: "esm",
+    ...sharedOptions,
+  });
+
+  // Remove the types file from the dist-src folder
+  const typeFiles = glob([
+    "./pkg/dist-src/types.js",
+    "./pkg/dist-src/types.js.map",
+    "./pkg/dist-src/**/types.js.map",
+    "./pkg/dist-src/**/types.js",
+  ]);
+  for (const typeFile of typeFiles) {
+    rmSync(typeFile);
+  }
+
+  const entryPoints = ["./pkg/dist-src/index.js"];
+
+  // Build the source code as a bundle for Node.js as CJS
+  await esbuild.build({
+    entryPoints,
+    outdir: "pkg/dist-node",
+    bundle: true,
+    platform: "node",
+    target: "node14",
```

Comment:
💅🏽 It would be nice to correlate this with `packageJson.engines.node`, but we can tackle this in a different PR.

```suggestion
    target: pkgJson.engines.node,
```

Reply:
It would be nice, but the `engines` field isn't just a node version, it's a semver expression

Reply:
True. It's not an easy task.